### PR TITLE
Add macOS arm64 to Cirrus CI

### LINKED ISSUE #12599: [Feature request] Add macOS arm64 to CI
### Quick summary

Add macOS arm64 to Cirrus CI

### Details

Cirrus CI currently only provides x86_64 macOS builds. 

~~As of writing, compilation for arm64 is broken (not just on Macs). See #12577~~ Edit: fixed 

There are occasionally regressions that are introduced that go unnoticed since there are no Arm CI builds. This can be prevented with a Cirrus test build. 

One argument that may be given for not doing this is that arm64 support on macOS is not fully implemented. However several 2D games such as Dragon's Crown and some of the Arkedo series do actually work.

Edit: Cirrus CI now builds the x86 macOS artefacts using an Arm64 VM, so cross-compilation is not an issue.

### PR DESCRIPTION
**Pull request by DarthMDev**
Attempts to Resolve #12599

### CODE REVIEW

**Comment by DarthMDev** [comment]
dont know why linux is failing 🤔

**Comment by DarthMDev** [comment]
I will squash all these commits when I am done testing and fixing.

File: .ci/build-mac-arm64.sh [hunk 1]
```diff
@@ -0,0 +1,62 @@
+#!/bin/sh -ex
+# llvm@16 doesnt exist for arm64 we will use the latest available
+brew install -f --overwrite nasm ninja git p7zip create-dmg ccache pipenv llvm@13 sdl2 glew cmake
+brew link -f llvm@13
```
[the remark this answers is not in the document]
Comment (DarthMDev) [reply .ci/build-mac-arm64.sh:4]: I tried , it failed

File: .ci/build-mac-arm64.sh [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+#export MACOSX_DEPLOYMENT_TARGET=12.0
+export CXX=clang++
+export CC=clang
+
+export BREW_PATH;
+BREW_PATH="$(brew --prefix)"
+export BREW_BIN="/usr/local/bin"
```
Comment (shinra-electric) [inline .ci/build-mac-arm64.sh:11]: Since this is the arm build, the brew path is opt/homebrew

Comment (DarthMDev) [reply .ci/build-mac-arm64.sh:11]: Thank you , i mostly use macports (lets me install universally ) so i forgot 😅

File: .ci/build-mac-arm64.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (DarthMDev) [reply .ci/build-mac-arm64.sh:4]: Warning: No available formula with the name "llvm@16". Did you mean llvm, llvm@13, llvm@12, llvm@11, llvm@9, llvm@8 or llvm@7?

[the remark this answers is not in the document]
Comment (shinra-electric) [reply .ci/build-mac-arm64.sh:4]: That’s weird. It should find llvm@16.  that’s how I build it..

File: .ci/build-mac-arm64.sh [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+mkdir build && cd build || exit 1
+
+"$BREW_PATH/bin/cmake" .. \
+    -DUSE_SDL=ON -DUSE_DISCORD_RPC=OFF -DUSE_VULKAN=ON -DUSE_ALSA=OFF -DUSE_PULSE=OFF -DUSE_AUDIOUNIT=ON \
+    -DLLVM_CCACHE_BUILD=OFF -DLLVM_BUILD_RUNTIME=OFF -DLLVM_BUILD_TOOLS=OFF \
+    -DLLVM_INCLUDE_DOCS=OFF -DLLVM_INCLUDE_EXAMPLES=OFF -DLLVM_INCLUDE_TESTS=OFF -DLLVM_INCLUDE_TOOLS=OFF \
+    -DLLVM_INCLUDE_UTILS=OFF -DLLVM_USE_PERF=OFF -DLLVM_ENABLE_Z3_SOLVER=OFF \
+    -DUSE_NATIVE_INSTRUCTIONS=OFF \
+    -DUSE_SYSTEM_MVK=OFF \
```
Comment (shinra-electric) [inline .ci/build-mac-arm64.sh:48]: You should also use `-DPNG_ARM_NEON=on`

**Comment by shinra-electric** [comment]
If it helps, this I what I do to compile locally:

```
brew install molten-vk vulkan-headers sdl2 nasm qt@5 ninja cmake glew ffmpeg llvm@16
# Make sure glslang and qt@6 is uninstalled, otherwise building will fail

export Qt5_DIR=$(brew --prefix)/opt/qt5
export VULKAN_SDK=$(brew --prefix)/opt/molten-vk
ln -s "$VULKAN_SDK/lib/libMoltenVK.dylib" "$VULKAN_SDK/lib/libvulkan.dylib"
export VK_ICD_FILENAMES=$VULKAN_SDK/share/vulkan/icd.d/MoltenVK_icd.json
export LLVM_DIR=$(brew --prefix)/opt/llvm@16

git clone --recurse-submodules https://github.com/RPCS3/rpcs3
cd rpcs3

sed -i '' "s/extern const double NSAppKitVersionNumber;/const double NSAppKitVersionNumber = 1343;/g" 3rdparty/hidapi/hidapi/mac/hid.c

mkdir build && cd build
cmake .. -DPNG_ARM_NEON=on -DUSE_ALSA=OFF -DUSE_PULSE=OFF -DUSE_AUDIOUNIT=ON -DUSE_NATIVE_INSTRUCTIONS=OFF -DUSE_SYSTEM_FFMPEG=on -DCMAKE_OSX_ARCHITECTURES="arm64" -DLLVM_TARGETS_TO_BUILD="AArch64;ARM" -GNinja -DUSE_SYSTEM_MVK=on -DUSE_FAUDIO=OFF

# -DUSE_FAUDIO=OFF
# This will prevent a warning after compiling

ninja && codesign --force --deep --sign - bin/rpcs3.app/Contents/MacOS/rpcs3
```

**Comment by DarthMDev** [comment]
Everything else has been uppercase apparently neon has to be lowercase 😂

File: .ci/build-mac-arm64.sh [hunk 4]
```diff
@@ -11,19 +11,11 @@ BREW_PATH="$(brew --prefix)"
...
-VK_ICD_FILENAMES="$VULKAN_SDK/share/vulkan/icd.d/MoltenVK_icd.json"
 export Qt5_DIR
 Qt5_DIR="$(brew --prefix)/opt/qt5"
-export SDL2_DIR
-SDL2_DIR="$BREW_PATH/opt/sdl2/lib/cmake/SDL2"
-export LLVM_DIR
-LLVM_DIR="$(brew --prefix)/opt/llvm@16"
+export PATH="$BREW_PATH/opt/llvm@16/bin:$BREW_BIN:$BREW_SBIN:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Library/Apple/usr/bin:$PATH"
+export LDFLAGS="-L$BREW_PATH/lib -Wl,-rpath,$BREW_PATH/lib"
+export CPPFLAGS="-I$BREW_PATH/include -msse -msse2 -mcx16 -no-pie"
```
Comment (shinra-electric) [inline .ci/build-mac-arm64.sh:18]: I don’t really understand things to this depth, but I would have thought that there are no SSE or SSE2 or any other x86 extensions on arm…?

Comment (DarthMDev) [reply .ci/build-mac-arm64.sh:18]: I don't fully understand it either im just experimenting

File: .ci/deploy-mac.sh [hunk 5]
```diff
@@ -15,7 +15,7 @@ echo "AVVER=$AVVER" >> ../.ci/ci-vars.env
 cd bin
 mkdir "rpcs3.app/Contents/lib/"
 
-cp "/usr/local/Homebrew/opt/llvm@16/lib/c++/libc++abi.1.0.dylib" "rpcs3.app/Contents/lib/libc++abi.1.dylib"
+cp "/usr/local/opt/llvm@16/lib/c++/libc++abi.1.0.dylib" "rpcs3.app/Contents/lib/libc++abi.1.dylib"
```
Comment (shinra-electric) [inline .ci/deploy-mac.sh:18]: This location doesn't exist on arm

Comment (shinra-electric) [inline .ci/deploy-mac.sh:18]:
```suggestion
cp "/opt/homebrew/opt/llvm@16/lib/c++/libc++abi.1.0.dylib" "rpcs3.app/Contents/lib/libc++abi.1.dylib"
```

Comment (shinra-electric) [inline .ci/deploy-mac.sh:18]:
```suggestion
ARCH=$(uname -p)
if [ $ARCH = "arm" ]; then
	cp "/opt/homebrew/opt/llvm@16/lib/c++/libc++abi.1.0.dylib" "rpcs3.app/Contents/lib/libc++abi.1.dylib"
else
	cp "/usr/local/opt/llvm@16/lib/c++/libc++abi.1.0.dylib" "rpcs3.app/Contents/lib/libc++abi.1.dylib"
fi
```

Comment (shinra-electric) [reply .ci/deploy-mac.sh]: I added double quotes around the `$CMAKE_EXTRA_OPS` and `$ARCH` as advised by the compiler. It should still work and will fix the linux builds

Comment (DarthMDev) [reply .ci/deploy-mac.sh]: Thank you

Comment (DarthMDev) [reply .ci/deploy-mac.sh]: Hmm it wont let me commit it

Comment (shinra-electric) [reply .ci/deploy-mac.sh]: I’m an idiot. We should be using `$(brew --prefix)` here

Comment (DarthMDev) [reply .ci/deploy-mac.sh]: oh yeah bc it will be different based on using rosetta2 or not

Comment (DarthMDev) [reply .ci/deploy-mac.sh]: its doing that dumb thing where it still wont let me commit ur suggestion ;-;

Comment (DarthMDev) [reply .ci/deploy-mac.sh]: "outdated suggestions cannot be applied"

Comment (shinra-electric) [reply .ci/deploy-mac.sh]: Maybe just close this comment and apply it manually?

Comment (shinra-electric) [inline .ci/deploy-mac.sh]:
```suggestion
cp "cp "$(brew --prefix)/opt/llvm@16/lib/c++/libc++abi.1.0.dylib" "rpcs3.app/Contents/lib/libc++abi.1.dylib"
```

File: .ci/build-mac-arm64.sh [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+export CPPFLAGS="-I$BREW_PATH/include"
+export LIBRARY_PATH="$BREW_PATH/lib"
+export LD_LIBRARY_PATH="$BREW_PATH/lib"
+export Qt5_DIR="/opt/homebrew/opt/qt@5/lib/cmake/Qt5"
+export VULKAN_SDK
+VULKAN_SDK="$(brew --prefix)/opt/molten-vk"
+ln -s "$VULKAN_SDK/lib/libMoltenVK.dylib" "$VULKAN_SDK/lib/libvulkan.dylib"
+export VK_ICD_FILENAMES="$VULKAN_SDK/share/vulkan/icd.d/MoltenVK_icd.json"
+export LLVM_DIR
+LLVM_DIR="$(brew --prefix)/opt/llvm@16"
```
Comment (UltraHDR) [inline .ci/build-mac-arm64.sh:27]:
```suggestion
LLVM_DIR="$(brew --prefix llvm@16)"
```

File: .ci/build-mac-arm64.sh [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+export BREW_SBIN="/opt/homebrew/sbin"
+export CMAKE_EXTRA_OPTS='-DLLVM_TARGETS_TO_BUILD=AArch64;ARM'
+export PATH="$BREW_PATH/opt/llvm@16/bin:$BREW_BIN:$BREW_SBIN:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Library/Apple/usr/bin:$PATH"
+export LDFLAGS="-L$BREW_PATH/lib -Wl,-rpath,$BREW_PATH/lib"
+export CPPFLAGS="-I$BREW_PATH/include"
+export LIBRARY_PATH="$BREW_PATH/lib"
+export LD_LIBRARY_PATH="$BREW_PATH/lib"
+export Qt5_DIR="/opt/homebrew/opt/qt@5/lib/cmake/Qt5"
+export VULKAN_SDK
+VULKAN_SDK="$(brew --prefix)/opt/molten-vk"
```
Comment (UltraHDR) [inline .ci/build-mac-arm64.sh:23]:
```suggestion
VULKAN_SDK="$(brew --prefix molten-vk)"
```

File: .ci/build-mac-arm64.sh [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+brew link -f llvm@16
+
+
+#export MACOSX_DEPLOYMENT_TARGET=12.0
+export CXX=clang++
+export CC=clang
+
+export BREW_PATH;
+BREW_PATH="$(brew --prefix)"
+export BREW_BIN="/opt/homebrew/bin"
```
Comment (UltraHDR) [inline .ci/build-mac-arm64.sh:13]:
```suggestion
export BREW_BIN="$BREW_PATH/bin"
```

File: .ci/build-mac-arm64.sh [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+export BREW_SBIN="/opt/homebrew/sbin"
```
Comment (UltraHDR) [inline .ci/build-mac-arm64.sh:14]:
```suggestion
export BREW_SBIN="$BREW_PATH/sbin"
```

File: .ci/build-mac-arm64.sh [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+export CMAKE_EXTRA_OPTS='-DLLVM_TARGETS_TO_BUILD=AArch64;ARM'
+export PATH="$BREW_PATH/opt/llvm@16/bin:$BREW_BIN:$BREW_SBIN:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Library/Apple/usr/bin:$PATH"
```
Comment (UltraHDR) [inline .ci/build-mac-arm64.sh:16]:
```suggestion
export PATH="$(brew --prefix llvm@16)/bin:$BREW_BIN:$BREW_SBIN:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Library/Apple/usr/bin:$PATH"
```

File: .ci/build-mac-arm64.sh [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+export LDFLAGS="-L$BREW_PATH/lib -Wl,-rpath,$BREW_PATH/lib"
+export CPPFLAGS="-I$BREW_PATH/include"
+export LIBRARY_PATH="$BREW_PATH/lib"
+export LD_LIBRARY_PATH="$BREW_PATH/lib"
+export Qt5_DIR="/opt/homebrew/opt/qt@5/lib/cmake/Qt5"
```
Comment (UltraHDR) [inline .ci/build-mac-arm64.sh:21]:
```suggestion
export Qt5_DIR="$(brew --prefix qt@5)/lib/cmake/Qt5"
```

**Comment by DarthMDev** [comment]
anyone have any ideas whats going on with homebrew to cause
Error: Fetching /opt/homebrew/Library/Taps/homebrew/homebrew-core failed!

File: .ci/build-mac-arm64.sh [hunk 12]
```diff
@@ -0,0 +1,70 @@
+#!/bin/sh -ex
+
+brew update
+brew install -f --overwrite nasm ninja git p7zip create-dmg ccache pipenv llvm@16 sdl2 glew cmake faudio qt@5 ffmpeg@6 vulkan-headers
```
Comment (shinra-electric) [inline .ci/build-mac-arm64.sh:4]:
```suggestion
brew install -f --overwrite nasm ninja git p7zip create-dmg ccache pipenv llvm@16 sdl2 glew cmake faudio qt@6 ffmpeg@6 vulkan-headers
```

File: .ci/build-mac-arm64.sh [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+export CMAKE_EXTRA_OPTS='-DLLVM_TARGETS_TO_BUILD=AArch64;ARM'
+
+export PATH
+PATH="$(brew --prefix llvm@16)/bin:$BREW_BIN:$BREW_SBIN:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Library/Apple/usr/bin:$PATH"
+export LDFLAGS="-L$BREW_PATH/lib -Wl,-rpath,$BREW_PATH/lib"
+export CPPFLAGS="-I$BREW_PATH/include"
+export LIBRARY_PATH="$BREW_PATH/lib"
+export LD_LIBRARY_PATH="$BREW_PATH/lib"
+export Qt5_DIR
+Qt5_DIR="$(brew --prefix qt@5)/lib/cmake/Qt5"
```
Comment (shinra-electric) [inline .ci/build-mac-arm64.sh:28]:
```suggestion
Qt6_DIR="$(brew --prefix qt@6)/lib/cmake/Qt6"
```

File: .cirrus.yml [hunk 14]
```diff
@@ -136,6 +136,15 @@ macos_task:
...
         - chmod +x ".ci/deploy-mac.sh"
         - ".ci/build-mac.sh"
+    - name: Cirrus macOS arm64
+      macos_instance:
+        image: ghcr.io/cirruslabs/macos-monterey-xcode:14
+      mac_script:
+        - mkdir artifacts
+        - chmod +x ".ci/build-mac-arm64.sh"
+        - chmod +x ".ci/deploy-mac.sh"
+        - ".ci/build-mac-arm64.sh"
```
Comment (Margen67) [inline .cirrus.yml:146]:
```suggestion
        - mkdir -p artifacts
        - chmod +x .ci/build-mac-arm64.sh .ci/deploy-mac.sh
        - .ci/build-mac-arm64.sh
```

**Comment by shinra-electric** [comment]
I'm getting a library not loaded error, even though the dylib is packaged. 

```
Termination Reason:    Namespace DYLD, Code 1 Library missing
Library not loaded: @loader_path/../../../../opt/sdl2/lib/libSDL2-2.0.0.dylib
Referenced from: <E87FD9F9-BBD6-322F-9670-CEA913D8DC25> /Volumes/VOLUME/*/RPCS3.app/Contents/Frameworks/libFAudio.0.dylib
```

**Comment by shinra-electric** [comment]
Actually, since the problem is FAudio dylib, why not just disable it with `-DUSE_FAUDIO=OFF`? 

FAudio is ignored in Windows builds (see #14128) , and it's not necessary for macOS (Cubeb works fine).

File: .ci/build-mac-arm64.sh [hunk 15]
```diff
@@ -0,0 +1,70 @@
+#!/bin/sh -ex
+
+brew update
+brew install -f --overwrite nasm ninja git p7zip create-dmg ccache pipenv llvm@16 sdl2 glew cmake faudio qt@6 ffmpeg@6 vulkan-headers
```
Comment (shinra-electric) [inline .ci/build-mac-arm64.sh:4]:
```suggestion
brew install -f --overwrite nasm ninja git p7zip create-dmg ccache pipenv llvm@16 sdl2 glew cmake qt@6 ffmpeg@6 vulkan-headers
```

File: .ci/build-mac-arm64.sh [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+export VULKAN_SDK
+VULKAN_SDK="$(brew --prefix molten-vk)"
+ln -s "$VULKAN_SDK/lib/libMoltenVK.dylib" "$VULKAN_SDK/lib/libvulkan.dylib"
+export VK_ICD_FILENAMES="$VULKAN_SDK/share/vulkan/icd.d/MoltenVK_icd.json"
+export LLVM_DIR
+LLVM_DIR="$(brew --prefix llvm@16)"
+
+# exclude FAudio, SPIRV, SDL , ffmpeg and LLVM from submodule update
+# shellcheck disable=SC2046
+git submodule -q update --init --depth=1 --jobs=8 $(awk '/path/ && !/FAudio/ && !/llvm/ && !/SPIRV/ && !/SDL/ && !/ffmpeg/ { print $3 }' .gitmodules)
```
Comment (shinra-electric) [inline .ci/build-mac-arm64.sh:39]:
```suggestion
git submodule -q update --init --depth=1 --jobs=8 $(awk '/path/ && !/llvm/ && !/SPIRV/ && !/SDL/ && !/ffmpeg/ { print $3 }' .gitmodules)
```

File: .ci/build-mac-arm64.sh [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+
+mkdir build && cd build || exit 1
+
+"$BREW_PATH/bin/cmake" .. \
+    -DUSE_SDL=ON -DUSE_DISCORD_RPC=OFF -DUSE_VULKAN=ON -DUSE_ALSA=OFF -DUSE_PULSE=OFF -DUSE_AUDIOUNIT=ON \
+    -DLLVM_CCACHE_BUILD=OFF -DLLVM_BUILD_RUNTIME=OFF -DLLVM_BUILD_TOOLS=OFF \
+    -DLLVM_INCLUDE_DOCS=OFF -DLLVM_INCLUDE_EXAMPLES=OFF -DLLVM_INCLUDE_TESTS=OFF -DLLVM_INCLUDE_TOOLS=OFF \
+    -DLLVM_INCLUDE_UTILS=OFF -DLLVM_USE_PERF=OFF -DLLVM_ENABLE_Z3_SOLVER=OFF \
+    -DUSE_NATIVE_INSTRUCTIONS=OFF \
+    -DUSE_SYSTEM_FAUDIO=ON \
```
Comment (shinra-electric) [inline .ci/build-mac-arm64.sh:52]:
```suggestion
    -DUSE_SYSTEM_FAUDIO=OFF \
```

File: .ci/deploy-mac.sh [hunk 18]
```diff
@@ -15,8 +15,31 @@ echo "AVVER=$AVVER" >> ../.ci/ci-vars.env
...
+	install_name_tool -change "/opt/homebrew/Cellar/ffmpeg/6.0/lib/libavformat.60.dylib" "@executable_path/../Frameworks/libavformat.60.dylib" "rpcs3.app/Contents/MacOS/rpcs3"
+	install_name_tool -change "/opt/homebrew/Cellar/ffmpeg/6.0/lib/libavutil.58.dylib" "@executable_path/../Frameworks/libavutil.58.dylib" "rpcs3.app/Contents/MacOS/rpcs3"
+	install_name_tool -change "/opt/homebrew/Cellar/ffmpeg/6.0/lib/libswresample.4.dylib" "@executable_path/../Frameworks/libswresample.4.dylib" "rpcs3.app/Contents/MacOS/rpcs3"
+	install_name_tool -change "/opt/homebrew/Cellar/ffmpeg/6.0/lib/libswscale.7.dylib" "@executable_path/../Frameworks/libswscale.7.dylib" "rpcs3.app/Contents/MacOS/rpcs3"
 
+else
+	cp "/usr/local/opt/llvm@16/lib/c++/libc++abi.1.0.dylib" "rpcs3.app/Contents/lib/libc++abi.1.dylib"
+
+	# change the path to app for sdl2
+	install_name_tool -change "/opt/homebrew/opt/sdl2/lib/libSDL2-2.0.0.dylib" "@executable_path/../Frameworks/libSDL2-2.0.0.dylib" "rpcs3.app/Contents/MacOS/rpcs3"
```
Comment (shinra-electric) [inline .ci/deploy-mac.sh:37]: This is supposed to be x86
```suggestion
	install_name_tool -change "/usr/local/opt/sdl2/lib/libSDL2-2.0.0.dylib" "@executable_path/../Frameworks/libSDL2-2.0.0.dylib" "rpcs3.app/Contents/MacOS/rpcs3"
```

File: .ci/deploy-mac.sh [hunk 19]
```diff
@@ -15,8 +15,31 @@ echo "AVVER=$AVVER" >> ../.ci/ci-vars.env
...
+	
+	# now faudio 
+	install_name_tool -change "/opt/homebrew/opt/faudio/lib/libFAudio.0.dylib" "@executable_path/../Frameworks/libFAudio.0.dylib" "rpcs3.app/Contents/MacOS/rpcs3"
```
Comment (shinra-electric) [inline .ci/deploy-mac.sh:40]: Same
```suggestion
	install_name_tool -change "/usr/local/opt/faudio/lib/libFAudio.0.dylib" "@executable_path/../Frameworks/libFAudio.0.dylib" "rpcs3.app/Contents/MacOS/rpcs3"
```

File: .ci/deploy-mac.sh [hunk 20]
```diff
@@ -15,8 +15,31 @@ echo "AVVER=$AVVER" >> ../.ci/ci-vars.env
...
+	install_name_tool -change "/opt/homebrew/Cellar/ffmpeg/6.0/lib/libavdevice.60.dylib" "@executable_path/../Frameworks/libavdevice.60.dylib" "rpcs3.app/Contents/MacOS/rpcs3"
+	install_name_tool -change "/opt/homebrew/Cellar/ffmpeg/6.0/lib/libavfilter.7.dylib" "@executable_path/../Frameworks/libavfilter.7.dylib" "rpcs3.app/Contents/MacOS/rpcs3"
+	install_name_tool -change "/opt/homebrew/Cellar/ffmpeg/6.0/lib/libavformat.60.dylib" "@executable_path/../Frameworks/libavformat.60.dylib" "rpcs3.app/Contents/MacOS/rpcs3"
+	install_name_tool -change "/opt/homebrew/Cellar/ffmpeg/6.0/lib/libavutil.58.dylib" "@executable_path/../Frameworks/libavutil.58.dylib" "rpcs3.app/Contents/MacOS/rpcs3"
+	install_name_tool -change "/opt/homebrew/Cellar/ffmpeg/6.0/lib/libswresample.4.dylib" "@executable_path/../Frameworks/libswresample.4.dylib" "rpcs3.app/Contents/MacOS/rpcs3"
+	install_name_tool -change "/opt/homebrew/Cellar/ffmpeg/6.0/lib/libswscale.7.dylib" "@executable_path/../Frameworks/libswscale.7.dylib" "rpcs3.app/Contents/MacOS/rpcs3"
```
Comment (shinra-electric) [inline .ci/deploy-mac.sh:32]: This needs to be done for arm too. 
```suggestion
	install_name_tool -change "/opt/homebrew/opt/faudio/lib/libFAudio.0.dylib" "@executable_path/../Frameworks/libFAudio.0.dylib" "rpcs3.app/Contents/MacOS/rpcs3"
```

File: .ci/build-mac-arm64.sh [hunk 21]
```diff
@@ -0,0 +1,70 @@
...
+# shellcheck disable=SC2046
+git submodule -q update --init --depth=1 --jobs=8 $(awk '/path/ && !/FAudio/ && !/llvm/ && !/SPIRV/ && !/SDL/ && !/ffmpeg/ { print $3 }' .gitmodules)
+
+# 3rdparty fixes
+sed -i '' "s/extern const double NSAppKitVersionNumber;/const double NSAppKitVersionNumber = 1343;/g" 3rdparty/hidapi/hidapi/mac/hid.c
+
+mkdir build && cd build || exit 1
+
+"$BREW_PATH/bin/cmake" .. \
+    -DUSE_SDL=ON -DUSE_DISCORD_RPC=OFF -DUSE_VULKAN=ON -DUSE_ALSA=OFF -DUSE_PULSE=OFF -DUSE_AUDIOUNIT=ON \
```
Comment (shinra-electric) [inline .ci/build-mac-arm64.sh:47]: Turning off FAudio resolves an issue with homebrew. Nas had to do the same thing on master.
```suggestion
    -DUSE_SDL=ON -DUSE_DISCORD_RPC=OFF -DUSE_VULKAN=ON -DUSE_ALSA=OFF -DUSE_PULSE=OFF -DUSE_FAUDIO=OFF -DUSE_AUDIOUNIT=ON \
```

File: .ci/build-mac-arm64.sh (hunk 15 above)
Comment (shinra-electric) [inline .ci/build-mac-arm64.sh:4]: Turning off FAudio resolves an issue with homebrew. Nas had to do the same thing on master.
```suggestion
brew install -f --overwrite nasm ninja git p7zip create-dmg ccache pipenv llvm@16 sdl2 glew cmake qt@6 ffmpeg@6 vulkan-headers
```

File: .ci/build-mac-arm64.sh (hunk 16 above)
Comment (shinra-electric) [inline .ci/build-mac-arm64.sh:39]: Remove FAudio for same reasons.
```suggestion
git submodule -q update --init --depth=1 --jobs=8 $(awk '/path/ && !/llvm/ && !/SPIRV/ && !/SDL/ && !/ffmpeg/ { print $3 }' .gitmodules)
```

File: .ci/build-mac-arm64.sh (hunk 17 above)
Comment (shinra-electric) [inline .ci/build-mac-arm64.sh:52]: Disable FAudio for same reasons
```suggestion
    -DUSE_SYSTEM_FAUDIO=OFF \
```

**Comment by shinra-electric** [comment]
Maybe have a look at what Nas did in #14563, that PR fixes a similar issue on master.

**Comment by hcorion** [comment]
FYI we've moved away from Cirrus for our CI (other than FreeBSD) because of their new limits.

**Comment by AniLeo** [comment]
Superseded and concluded by #16070

**Pull request closed without merging**